src/Implement/RequestHandler.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "RequestHandler.h"
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+
+std::unordered_map<std::string, APIHandler *> RequestHandler::apis;
+
+RequestHandler::RequestHandler() {
+  
+}
+
+void RequestHandler::SetFd(const int &file_descriptor){
+  this->client_fd = file_descriptor;
+}
+
+void RequestHandler::WaitForMessage() {
+  std::string tmp_string = "";
+  char input_buffer[4096] = {};
+  std::string next_request = "";
+  // only read header if message is bigger than buffer
+  while (true) {
+    int function_flag =
+        recv(this->client_fd, input_buffer, sizeof(input_buffer), 0);
+    if (function_flag > 0) {
+      std::string to_str = next_request + input_buffer;
+      std::string find_key = "Content-Length:";
+      size_t start_cut_pos = to_str.find(find_key, 0) + find_key.length();
+      size_t end_cut_pos = to_str.find("\r\n\r\n", 0);
+      // keep adding message until find Content-Length
+      if (to_str.find(find_key, 0) != std::string::npos) {
+        // get content length
+        int client_content_length = std::stoi(
+            to_str.substr(start_cut_pos, end_cut_pos - start_cut_pos));
+        tmp_string += to_str.substr(0, end_cut_pos + 4 + client_content_length);
+        next_request = to_str.substr(
+            end_cut_pos + 4 + client_content_length,
+            to_str.length() - (end_cut_pos + 4 + client_content_length));
+        // parse message to OperateData
+        RequestParser *tmp_parser = RequestParser::GetInstance();
+        this->data = tmp_parser->ParseData(tmp_string);
+        this->data->header["client_file_descriptor"] =
```

Comment:
This is a very hack-ish way to pass file descriptors, it is recommended that you pass your file descriptors as argument to the function call, not as header.